src/parsers/psse_dynamic_data.jl

```diff
@@ -0,0 +1,267 @@
+"""
+Parse .dyr file into a dictionary indexed by bus number.
+Each bus number key has a dictionary indexed by component type and id.
+
+"""
+function _parse_dyr_file(file::AbstractString)
+    dyr_text = read(file, String)
+    start = 1
+    parsed_values = Dict{Int, Dict}()
+    models = Vector{String}()
+    while start < length(dyr_text)
+        val = findnext('/', dyr_text, start)
+        if isnothing(val)
+            break
+        end
+        line = replace(dyr_text[start:(val - 1)], "\'" => "")
+        val_array = strip.(split(line))
+        bus = parse(Int, val_array[1])
+        model = string(val_array[2])
+        id = parse(Int, val_array[3])
+        component_dict = get!(parsed_values, bus, Dict{Tuple{String, Int}, Array}())
+        component_dict[(model, id)] = parse.(Float64, val_array[4:end])
+        start = val + 1
+    end
+    return parsed_values
+end
+
+"""
+Populate arguments in a vector for each dynamic component (except Shafts).
+Returns a vector with the parameter values of the argument of each component.
+
+"""
+function _populate_args(param_map, val)
+    struct_args = Vector{Any}(undef, length(param_map))
+    for (ix, _v) in enumerate(param_map)
+        #If the parameter is a Float64, then use the value directly as the argument.
+        #Typically uses for the resistance that is not available in .dyr files.
+        if isa(_v, Float64)
+            struct_args[ix] = _v
+            #If the parameter is an Int64, then use the integer as the key of the value in the dictionary.
+        elseif isa(_v, Int64)
+            struct_args[ix] = val[_v]
+            #If the parameter is a tuple (as a string), then construct the tuple directly.
+        else
+            _t = strip(_v, ['(', ')'])
+            _t2int = parse.(Int64, split(_t, ','))
+            struct_args[ix] = (val[_t2int[1]], val[_t2int[2]])
+        end
+    end
+    return struct_args
+end
+
+"""
+Create a SingleMass shaft struct directly using the parameter mapping.
+
+"""
+function _make_shaft(param_map, val)
+    constructor_shaft =
+        (args...) -> InteractiveUtils.getfield(PowerSystems, :SingleMass)(args...)
+    shaft_args = _populate_args(param_map, val)
+    return constructor_shaft(shaft_args...)
+end
+
+function _make_source(g::StaticInjection, r::Float64, x::Float64)
+    return Source(
+        name = get_name(g),
+        available = true,
+        bus = get_bus(g),
+        active_power = get_active_power(g),
+        reactive_power = get_reactive_power(g),
+        R_th = r,
+        X_th = x,
+    )
+end
+
+function _assign_missing_components!(bus_dict, component_table)
+    for va in values(bus_dict)
+        !isassigned(va, component_table["AVR"]) &&
```

Comment:
I think we've agreed in the coding standards to implement these as if statements.